src/adapters/Celer/CelerReceiverAdapter.sol

```diff
@@ -13,75 +18,126 @@ interface IMessageReceiverApp {
-     * @param _executor Address who called the MessageBus execution function
-     */
+    /// @notice Called by MessageBus to execute a message
+    /// @param _sender The address of the source app contract
+    /// @param _srcChainId The source chain ID where the transfer is originated from
+    /// @param _message Arbitrary message bytes originated from and encoded by the source app contract
+    /// @param _executor Address who called the MessageBus execution function
     function executeMessage(address _sender, uint64 _srcChainId, bytes calldata _message, address _executor)
         external
         payable
         returns (ExecutionStatus);
 }
 
-contract CelerReceiverAdapter is IBridgeReceiverAdapter, MessageAppPauser, IMessageReceiverApp {
-    mapping(uint256 => address) public senderAdapters;
+contract CelerReceiverAdapter is IBridgeReceiverAdapter, IMessageReceiverApp {
+    string constant ABORT_PREFIX = "MSG::ABORT:";
+
     address public immutable msgBus;
-    mapping(bytes32 => bool) public executedMessages;
+    IGAC public immutable gac;
+
+    /*/////////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    ////////////////////////////////////////////////////////////////*/
 
-    event SenderAdapterUpdated(uint256 srcChainId, address senderAdapter);
+    /// @dev adapter deployed to Ethereum
+    address public senderAdapter;
+    uint64 public senderChain;
+
+    /// @dev tracks the msg id status to prevent replay
+    mapping(bytes32 => bool) public isMessageExecuted;
+
+    /*/////////////////////////////////////////////////////////////////
+                                 MODIFIER
+    ////////////////////////////////////////////////////////////////*/
+    modifier onlyCaller() {
+        if (!gac.isPrevilagedCaller(msg.sender)) {
+            revert Error.INVALID_PREVILAGED_CALLER();
+        }
+        _;
+    }
 
     modifier onlyMessageBus() {
-        require(msg.sender == msgBus, "caller is not message bus");
+        if (msg.sender == msgBus) {
+            revert Error.CALLER_NOT_CELER_BUS();
+        }
         _;
     }
 
-    constructor(address _msgBus) {
+    /*/////////////////////////////////////////////////////////////////
+                                CONSTRUCTOR
+    ////////////////////////////////////////////////////////////////*/
+    constructor(address _msgBus, address _gac) {
         msgBus = _msgBus;
+        gac = IGAC(_gac);
     }
 
-    // Called by MessageBus on destination chain to receive cross-chain messages.
-    // The message is abi.encode of (MessageStruct.Message).
+    /*/////////////////////////////////////////////////////////////////
+                                EXTERNAL FUNCTIONS
+    ////////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IBridgeReceiverAdapter
+    function updateSenderAdapter(bytes memory _senderChain, address _senderAdapter) external override onlyCaller {
```

Comment:
we won't need to update the `_senderChain` in our case, so that parameter can be removed

Reply:
we can just make this an immutable variable that is set in the constructor right, without needing to hardcode